src/Client.php

```diff
@@ -47,14 +68,20 @@ class Client
      * @param array $httpOptions
      * @param ClientInterface $httpClient
      * @param string $requestMethod
+     * @param string $authType
      */
     public function __construct(
         string $endpointUrl,
         array $authorizationHeaders = [],
         array $httpOptions = [],
         ClientInterface $httpClient = null,
-        string $requestMethod = 'POST'
+        string $requestMethod = 'POST',
+        string $authType = self::AUTH_TYPE_HEADER
     ) {
+        if (!in_array($authType, self::ALLOWED_AUTH_TYPES, true)) {
+            throw new AuthTypeNotSupportedException($requestMethod);
+        }
+        $this->auth = new (self::AUTH_MAP[$authType]);
```

Comment:
Instead of having the user pass a string to the constructor and then we instantiate a class for it, I would recommend adding a nullable `AuthInterface` argument to the constructor, then the user would pass their own authenticator if they need it.